### ISSUE TITLE
Bug: allOf breaks mock generator

### ISSUE THREAD

**Issue by yekver** [anymaniax/orval#362]
```yaml
openapi: '3.0.0'
info:
  version: 1.0.0
  title: Swagger Petstore
  license:
    name: MIT
servers:
  - url: http://petstore.swagger.io/v1
paths:
  /pets:
    post:
      summary: Create a pet
      operationId: createPets
      responses:
        '200':
          description: Created Pet
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Pet'
components:
  schemas:
    Id:
      type: number
    Pet:
      type: object
      properties:
        someField:
          description: unexpected error
          allOf: # PAY ATTENTION TO THIS LINE
            - $ref: '#/components/schemas/Id'

```

produces invalid mock creator:
```javascript
// With allOf:
// ERROR: Parsing error: ',' expected
export const getCreatePetsMock = () => ({someField: faker.helpers.randomize([{faker.datatype.number()}, undefined])})

//Without allOf:
export const getCreatePetsMock = () => ({someField: faker.helpers.randomize([faker.datatype.number(), undefined])})
```

**Comment by cowofevil** [comment]
@yekver I ran into a similar issue, but I could not reproduce this bug with the OpenAPI spec you provided.

However, I was able to reproduce this bug with this spec: 

```YAML
openapi: 3.0.3
info:
  version: 1.0.0
  title: AllOf
paths:
  /workingendpoint:
    get:
      summary: An endpoint that uses a working demo of allOf
      operationId: workingEndpoint
      tags:
        - working
      responses:
        '200':
          description: Get working schema
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Working'
  /brokenendpoint:
    get:
      summary: An endpoint that uses a broken demo of allOf
      operationId: brokenEndpoint
      tags:
        - broken
      responses:
        '200':
          description: Get broken schema
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Broken'
components:
  schemas:
    CommonProperties:
      type: object
      description: The first set of common properties.
      properties:
        commonPropsField1:
          type: string
          description: 'Common property field #1'
        commonPropsField2:
          type: string
          description: 'Common property field #2'
    Working:
      allOf:
        - $ref: '#/components/schemas/CommonProperties'
        - type: object
          description: A working example of using allOf.
          properties:
            workingField:
              type: string
              description: 'Working field'
    Broken:
      allOf:
        - $ref: '#/components/schemas/CommonProperties'
        - type: object
          description: A broken example of using allOf.
```

The complete demonstration of the bug can be found here: https://github.com/cowofevil/orval-bug-362

**Comment by anymaniax** [comment]
fixed thanks for the examples @cowofevil & @yekver. Will be in the next release

**Comment by anymaniax** [comment]
Released with 6.9.0
